spec.emu

```diff
@@ -111,6 +111,32 @@ contributors: Ujjwal Sharma, Younies Mahmoud
       </table>
     </emu-table>
 
+    <emu-clause id="sec-todurationrecord" type="abstract operation">
+      <h1>
+        ToDurationRecord (
+          _input_: an Object
+        ): a Record with the fields listed in the Value Slot column of <emu-xref href="#table-duration-components"></emu-xref>
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It converts a given object that represents a Duration into a Record.</dd>
+      </dl>
+
+      <emu-alg>
+        1. Let _result_ be a new Record.
+        1. For each row in <emu-xref href="#table-duration-components"></emu-xref>, except the header row, in table order, do
+          1. Let _valueSlot_ be the Value Slot value.
+          1. Let _unit_ be the Unit value.
+          1. Let _value_ be ? Get(_input_, _unit_).
```

Comment:
If we are not doing string parsing, then we will need to handle the case where Type(_input_) is not Object. I guess it should throw a TypeError at the beginning of the operation.